src/main/resources/templates/index.html

```diff
@@ -1,21 +1,20 @@
 <!DOCTYPE html>
-<html lang="en" xmlns:th="http://www.thymeleaf.org">
+<html lang="en" xmlns:th="http://www.w3.org/1999/xhtml">
 <head>
     <meta charset="UTF-8">
-    <title>CodeFellowship</title>
+    <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>CodeFellowship - Home</title>
+    <link rel="stylesheet" href="/css/styles.css">
 </head>
 <body>
-<h1>Welcome to CodeFellowship</h1>
-<p>Connect with fellow code enthusiasts and share your journey.</p>
-
-<div th:if="${username}">
-    <p>Welcome, <span th:text="${username}"></span>!</p>
-    <a href="/logout">Logout</a>
-</div>
-
-<div th:unless="${username}">
-    <a href="/signup">Sign Up</a>
-    <a href="/login">Login</a>
-</div>
+<header>
+    <nav th:replace="fragments/navbar :: navbar"></nav>
```

Comment:
When you're importing fragments in Thymeleaf, make sure you're naming them exactly the same as the filename. You don't have any fragments called "navbar.html" or "error.html", but you do have fragments called "nav-header.html" and "error-fragment.html". You can either change the reference here in the template (and all of your other templates), or rename the nav-header and error-fragment files to match the variable name here.